src/screens/ExchangeModal.tsx

```diff
@@ -862,57 +881,120 @@ export default function ExchangeModal({
-        },
-        type: 'swap_details',
-      });
-      analytics.track('Opened Swap Details modal', {
-        inputTokenAddress: inputCurrency?.address || '',
-        inputTokenName: inputCurrency?.name || '',
-        inputTokenSymbol: inputCurrency?.symbol || '',
-        outputTokenAddress: outputCurrency?.address || '',
-        outputTokenName: outputCurrency?.name || '',
-        outputTokenSymbol: outputCurrency?.symbol || '',
-        type,
-      });
+  const navigateToSwapDetailsModal = useCallback(
+    (isRefuelTx = false) => {
+      android && Keyboard.dismiss();
+      const lastFocusedInputHandleTemporary = lastFocusedInputHandle.current;
+      android && (lastFocusedInputHandle.current = null);
+      inputFieldRef?.current?.blur();
+      outputFieldRef?.current?.blur();
+      nativeFieldRef?.current?.blur();
+      const internalNavigate = () => {
+        android && Keyboard.removeListener('keyboardDidHide', internalNavigate);
+        setParams({ focused: false });
+        navigate(Routes.SWAP_DETAILS_SHEET, {
+          confirmButtonProps,
+          currentNetwork,
+          flashbotTransaction: flashbots,
+          isRefuelTx,
+          restoreFocusOnSwapModal: () => {
+            android &&
+              (lastFocusedInputHandle.current = lastFocusedInputHandleTemporary);
+            setParams({ focused: true });
+          },
+          type: 'swap_details',
+        });
+        analytics.track('Opened Swap Details modal', {
+          inputTokenAddress: inputCurrency?.address || '',
+          inputTokenName: inputCurrency?.name || '',
+          inputTokenSymbol: inputCurrency?.symbol || '',
+          outputTokenAddress: outputCurrency?.address || '',
+          outputTokenName: outputCurrency?.name || '',
+          outputTokenSymbol: outputCurrency?.symbol || '',
+          type,
+        });
+      };
+      ios || !isKeyboardOpen()
+        ? internalNavigate()
+        : Keyboard.addListener('keyboardDidHide', internalNavigate);
+    },
+    [
+      confirmButtonProps,
+      currentNetwork,
+      flashbots,
+      inputCurrency?.address,
+      inputCurrency?.name,
+      inputCurrency?.symbol,
+      inputFieldRef,
+      lastFocusedInputHandle,
+      nativeFieldRef,
+      navigate,
+      outputCurrency?.address,
+      outputCurrency?.name,
+      outputCurrency?.symbol,
+      outputFieldRef,
+      setParams,
+      type,
+    ]
+  );
+
+  const navigateToRefuelModal = useCallback(() => {
+    const getNetworkDetails = (network: Network) => {
+      switch (network) {
+        case Network.arbitrum:
+          return {
+            gasToken: 'Arbitrum ETH',
+            name: 'Arbitrum',
+          };
+        case Network.optimism:
+          return {
+            gasToken: 'Optimism ETH',
+            name: 'Optimism',
+          };
+        case Network.polygon:
+          return {
+            gasToken: 'Polygon Matic',
+            name: 'Polygon',
+          };
+      }
```

Comment:
i remember this was in other file before or i'm mistaken?

Reply:
No it wasnt in another file, where do you think might be best to put it?

Reply:
maybe we can add gasToken here and then cut the switch statement

Reply:
https://github.com/rainbow-me/rainbow/blob/develop/src/helpers/networkInfo.ts